src/lib/services/types/proposal.ts

```diff
@@ -363,15 +230,54 @@ const zProposalAnswerCountsResponse = z
     }),
   })
   .transform(snakeToCamel);
-
 export type ProposalAnswerCountsResponse = z.infer<
   typeof zProposalAnswerCountsResponse
 >;
 
-export const getProposalAnswerCounts = async (
-  endpoint: string,
-  id: number
-): Promise<ProposalAnswerCountsResponse> =>
-  axios
-    .get(`${endpoint}/${encodeURIComponent(id)}/answer-counts`)
-    .then(({ data }) => parseWithError(zProposalAnswerCountsResponse, data));
+export const zProposalsResponseItemLcd = z
+  .object({
+    id: z.coerce.number(),
+    messages: z
+      .object({
+        "@type": z.string(),
+      })
+      .passthrough()
+      .array(),
+    status: z.string(),
+    final_tally_result: z.object({
+      yes_count: z.string(),
+      no_count: z.string(),
+      abstain_count: z.string(),
+      no_with_veto_count: z.string(),
+    }),
+    submit_time: zUtcDate,
+    deposit_end_time: zUtcDate,
+    total_deposit: zCoin.array(),
+    voting_start_time: zUtcDate,
+    voting_end_time: zUtcDate,
+    metadata: z.string(),
+    title: z.string(),
+    summary: z.string(),
+    proposer: zBechAddr,
+    expedited: z.boolean().optional().default(false),
+  })
+  .transform<Proposal>((val) => ({
+    ...snakeToCamel(val),
+    status: val.status
+      .replace("PROPOSAL_STATUS_", "")
+      .split("_")
+      .map((term: string) => capitalize(term.toLowerCase()))
+      .join("") as ProposalStatus,
+    resolvedHeight: null,
+    types: [],
```

Comment:
Let's derived types from messages by accessing to @type

Reply:
Correct?
`types: val.messages.map((msg) => msg["@type"]),`